torch/_inductor/decomposition.py

```diff
@@ -648,3 +655,40 @@ def masked_scatter(self, mask, source):
         source_idx = mask.reshape(-1).cumsum(0) - 1
         return inductor_prims.masked_scatter_with_index(self, mask, source_idx, source)
     return NotImplemented
+
+
+@out_wrapper()
+@register_decomposition(aten.avg_pool2d)
+def avg_pool2d(
+    x: torch.Tensor,
+    kernel_size: Union[Tuple[int, int], int],
+    stride: Optional[Union[Tuple[int, int], int]] = None,
+    padding: Union[Tuple[int, int], int] = 0,
+    ceil_mode: bool = False,
+    count_include_pad: bool = True,
+    divisor_override: Optional[int] = None,
+):
+    kernel_size = pad_listlike(kernel_size, 2)
+    window_size = functools.reduce(operator.mul, kernel_size)
+
+    # TODO: remove this when #100331 is merged. We only do this
+    # for window_size <=25 to avoid performance regressions compared
+    # to the previous algorithm which unrolled manually for <=25
+    context = (
+        config.patch(unroll_reductions_threshold=25)
+        if window_size <= 25
+        else contextlib.nullcontext()
+    )
+
+    with context:
```

Comment:
This is way too hacky. A more pythonic way would be:
```python
if window_size <= 25:
  # manually unroll via a for-loop
else:
  # x.sum()
```
In any case, you should discuss these decomps with @peterbell10 as we may need to think a bit about how we implement the decomp itself to make it fast in an A100. We had some ideas in a discussion today that may be worth trying.